projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func main() {
+	for {
+		weather, err := getWeatherWithRetries(3) // Retry up to 3 times
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+			os.Exit(1)
+		}
+
+		fmt.Println(weather)
+
+		// Sleep for a while before making the next request
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func getWeatherWithRetries(maxRetries int) (string, error) {
+	for i := 0; i < maxRetries; i++ {
+		weather, err := getWeather()
+		if err == nil {
+			return weather, nil
+		}
+
+		fmt.Fprintf(os.Stderr, "Failed to fetch weather: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Retrying (%d/%d)...\n", i+1, maxRetries)
+		time.Sleep(1 * time.Second) // Wait for a short duration before retrying
+	}
+
+	return "", fmt.Errorf("exceeded maximum retries")
```

Comment:
It may be nice to wrap one ore all of the errors encountered here, so we know why retrying was needed in the first place. For a user, knowing if an error looks like "You were offline" vs "The server was broken" may be useful to distinguish.